python/mxnet/rnn/rnn_cell.py

```diff
@@ -1068,3 +1069,348 @@ def unroll(self, length, inputs, begin_state=None, layout='NTC', merge_outputs=N
+    i2h_kernel : tuple of int
+        Kernel of Convolution operator in input-to-state transitions.
+    i2h_stride : tuple of int
+        Stride of Convolution operator in input-to-state transitions.
+    i2h_pad : tuple of int
+        Pad of Convolution operator in input-to-state transitions.
+    i2h_dilate : tuple of int
+        Dilation of Convolution operator in input-to-state transitions.
+    activation : str or Symbol,
+        Type of activation function.
+    prefix : str, default ''
+        Prefix for name of layers (and name of weight if params is None).
+    params : RNNParams, default None
+        Container for weight sharing between cells. Created if None.
+    conv_layout : str, , default 'NCHW'
+        Layout of ConvolutionOp
+    """
+    def __init__(self, input_shape, num_hidden,
+                 h2h_kernel, h2h_dilate,
+                 i2h_kernel, i2h_stride,
+                 i2h_pad, i2h_dilate,
+                 activation,
+                 prefix='', params=None, conv_layout='NCHW'):
+        super(BaseConvRNNCell, self).__init__(prefix=prefix, params=params)
+        # Convolution setting
+        self._h2h_kernel = h2h_kernel
+        assert (self._h2h_kernel[0] % 2 == 1) and (self._h2h_kernel[1] % 2 == 1), \
+            "Only support odd number, get h2h_kernel= %s" % str(h2h_kernel)
+        self._h2h_pad = (h2h_dilate[0] * (h2h_kernel[0] - 1) // 2,
+                         h2h_dilate[1] * (h2h_kernel[1] - 1) // 2)
+        self._h2h_dilate = h2h_dilate
+        self._i2h_kernel = i2h_kernel
+        self._i2h_stride = i2h_stride
+        self._i2h_pad = i2h_pad
+        self._i2h_dilate = i2h_dilate
+
+        self._num_hidden = num_hidden
+        self._input_shape = input_shape
+        self._conv_layout = conv_layout
+        self._activation = activation
+
+        # Infer state shape
+        data = symbol.Variable('data')
+        self._state_shape = symbol.Convolution(data=data,
+                                               num_filter=self._num_hidden,
+                                               kernel=self._i2h_kernel,
+                                               stride=self._i2h_stride,
+                                               pad=self._i2h_pad,
+                                               dilate=self._i2h_dilate,
+                                               layout=conv_layout)
+        self._state_shape = self._state_shape.infer_shape(data=input_shape)[1][0]
+        self._state_shape = (0, ) + self._state_shape[1:]
+
+    @property
+    def state_info(self):
+        return [{'shape': self._state_shape, '__layout__': self._conv_layout},
+                {'shape': self._state_shape, '__layout__': self._conv_layout}]
+
+    def __call__(self, inputs, states):
+        raise NotImplementedError("BaseConvRNNCell is abstract class for convolutional RNN")
+
+class ConvRNNCell(BaseConvRNNCell):
+    """Convolutional RNN cells
```

Comment:
In the future, we should also list the formula of ConvRNN layers. For ConvLSTM, the formula is like this:
```
i_t = \sigma(W_i \ast x_t + R_i \ast h_{t-1} + b_i)
f_t = \sigma(W_f \ast x_t + R_f \ast h_{t-1} + b_f)
o_t = \sigma(W_o \ast x_t + R_o \ast h_{t-1} + b_o)
c^\prime_t = tanh(W_c \ast x_t + R_c \ast h_{t-1} + b_c)
c_t = f_t \circ c_{t-1} + i_t \circ c^\prime_t
h_t = o_t \circ tanh(c_t)
```

Reply:
Hello, I just found these equations differ from those presented in the original [ConvLSTM](https://arxiv.org/pdf/1506.04214.pdf) paper in the computations of input gate and forget gate. I'm wondering if this formulation performs better than the original one by removing the contribution of C to i_t and f_t?

Reply:
Not only the ConvLSTMCell but  also LSTMCell has removed the contribution of C_{t-1} in the input gate, forget gate and output gate. 

But I add the Hadamard product about C_{t-1} for personal use.